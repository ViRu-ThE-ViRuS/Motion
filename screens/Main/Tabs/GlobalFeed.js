import React, { Component } from 'react';

import {
    StyleSheet,
    View,
    Text,
    BackHandler,
} from 'react-native';

import NavBarComponent from './../../../components/NavBarComponent/NavBarComponent';

export default class GlobalFeed extends Component {
    static navigationOptions = {
        title: 'Global Feed',
        headerRight: (<NavBarComponent />),
    }

    componentWillMount() {
        BackHandler.addEventListener('hardwareBackPress', () => this.props.navigation.navigate('local'));
    }

    componentWillUnmount() {
        BackHandler.removeEventListener('hardwareBackPress', BackHandler.exitApp);
    }

    render() {
        return (
          <View style={styles.container}>
            <Text>
                    Global
            </Text>
          </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {

    },
});
