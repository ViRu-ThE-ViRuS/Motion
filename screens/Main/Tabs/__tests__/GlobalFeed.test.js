import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import GlobalFeed from './../GlobalFeed';

describe('Global Feed tab tests', () => {
    let tree;

    beforeEach(() => {
        tree = renderer.create(
          <GlobalFeed />
        );
    });

    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });

    it('removes backhandler correctly', () => {
        expect(tree.getInstance().componentWillUnmount()).toBe(undefined);
    });
});
