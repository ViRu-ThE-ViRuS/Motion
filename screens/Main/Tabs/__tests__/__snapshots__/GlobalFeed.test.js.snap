// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Feed tab tests renders correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Global
  </Text>
</View>
`;
